### ISSUE TITLE
Enhanced scaling qdrant node startup process

### ISSUE THREAD

**Issue by janljd** [qdrant/qdrant#4980]
**Is your feature request related to a problem? Please describe.**
We started with a one node qdrant cluster on k8s and then scaled(under search upsert process) to 1 to 2, 2 to 6 nodes

**first expansion (1 to 2):** the usage process includes create, delete ,update collections, add replica , move shards etc 
**second expansion (2 to 6):** The k8s node status of the new node was ready quickly, but the server took a long time (about 1 hour) to do the collection_meta_ops/consensus_manager process, and most of the operations were useless, .e.g. the collection no longer exists
e.g.
```
INFO storage::content_manager::toc::collection_meta_ops: Deleting collection tl-TEST_VDB_entity    
INFO storage::content_manager::toc::collection_meta_ops: Creating collection tl-TEST_VDB_entity    
INFO storage::content_manager::toc::collection_meta_ops: Deleting collection tl-TEST_VDB_entity    
INFO storage::content_manager::toc::collection_meta_ops: Creating collection tl-TEST_VDB_entity    
NFO storage::content_manager::toc::collection_meta_ops: Deleting collection tl-TEST_VDB_entity    
INFO storage::content_manager::toc::collection_meta_ops: Creating collection tl-TEST_VDB_entity    
INFO storage::content_manager::toc::collection_meta_ops: Deleting collection tl-TEST_VDB_entity 

WARN storage::content_manager::consensus_manager: Failed to apply collection meta operation entry with user error: Bad request: There is no transfer for shard 0 from 4566347219847216 to 186469645116806    
WARN storage::content_manager::consensus_manager: Failed to apply collection meta operation entry with user error: Bad request: There is no transfer for shard 1 from 4566347219847216 to 186469645116806    
WARN storage::content_manager::consensus_manager: Failed to apply collection meta operation entry with user error: Bad request: There is no transfer for shard 0 from 4566347219847216 to 186469645116806    
WARN storage::content_manager::consensus_manager: Failed to apply collection meta operation entry with user error: Bad request: There is no transfer for shard 1 from 4566347219847216 to 18646964511680 
```
Before the synchronization process is completed, the new node will receive a new request, and a not found error will occur when the requested collection meta is not synchronized.

**Describe the solution you'd like**
When a new node is started, there will not be any shards on it. It should only synchronize the current collection meta.  During the process, the k8s node status remains Unavailable.

**Comment by timvisee** [comment]
Correct. We currently replay all consensus operations when a new node is joining.

We do have a mechanism to prevent this (consensus snapshots). However, this is still disabled because we aren't 100% certain they are robust and reliable. We'll pick them up again shortly which will resolve this issue.

The above case is easily reproducible:
1. Create a new cluster with one node, and create+delete a collection 100 times like this:
   ```bash
   for i in $(seq 100); do
       bfb -n 1 --shards 1 --segments 1 --skip-wait-index
   done
   ```
2. Bootstrap a second node
3. See it replay all the operations

**Comment by janljd** [comment]
Thanks

Is there any workarounds (e.g. manual cleanup  consensus operations logs) we can take to avoid this issue before new version release ?

**Comment by timvisee** [comment]
Currently, no.

Can I ask what you're using to generate so many operations? Are you creating a huge number of collections? I'm asking because we normally strongly recommend to create [a single collection for multiple tenants](https://qdrant.tech/documentation/guides/multiple-partitions/), it works much better.

**Comment by janljd** [comment]
thanks 
We will use a collection for multiple tenants in our next project

What are you using to generate so many operations?
Since we started using qdrant in a small environment (1 node and limited cpu) to support many projects (different vector sizes), we need to scale qdrant due to increased usage, so we need to copy/move the shards to a new node

**Comment by janljd** [comment]
[consensus-checkpointing](https://qdrant.tech/documentation/guides/distributed_deployment/?q=chec#consensus-checkpointing) 
We tried attaching the new node to the cluster after calling /cluster/recover api, but it didn't seem to help
